docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,306 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Agent Clustering enables Grafana Agent Flow to coordinate a fleet of agents to
+work together for workload distribution and high availability. It makes agent
+deployments horizontally scalable by default, and minimizes the operational
+overhead of managing your agent infrastructure to handle demand.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model and
+the lightweight [grafana/ckit][] framework. All running nodes are assumed to
+use the same configuration and have access to the same type of hardware and
```

Comment:
Maybe add somewhere later a bit about convergence around a config when rolling our new configurations, there is some of this in the troubleshooting but from reading this line to me it says I cant do a rolling release but have to do a full up and down when changing configuration.